### PR TITLE
fix: [PIE-3977]: Template service definition set in context for propagation usage

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

- Added `templateServiceDefinitionType` map in context along with templateTypes
- Consuming `templateServiceData` to get serviceDefinition in case of stage propagation

#### Screenshots

NA

https://user-images.githubusercontent.com/100121280/179232443-cc96456b-ef6f-4a55-9132-fe4665c16b2f.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineStudio/PipelineTemplateBuilder/TemplatePipelineCanvas/__tests__/TemplatePipelineCanvas.test.tsx [hunk 1]
```diff
@@ -114,7 +114,8 @@ jest.mock('@pipeline/utils/templateUtils', () => ({
   ...jest.requireActual('@pipeline/utils/templateUtils'),
   getTemplateTypesByRef: () =>
     Promise.resolve({
-      Test_Stage_Template: 'Deployment'
+      templateTypes: { Test_Stage_Template: 'Deployment' },
+      templateServiceData: {}
```
Comment (yc147) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineTemplateBuilder/TemplatePipelineCanvas/__tests__/TemplatePipelineCanvas.test.tsx:118]: Add test for this.

**Review by yc147 — APPROVED** [review]

File: src/modules/70-pipeline/utils/templateUtils.ts [hunk 2]
```diff
@@ -136,16 +144,22 @@ export const getTemplateTypesByRef = (
...
     .then(responses => {
+      const templateServiceData = {}
       const templateTypes = {}
       responses.forEach(response => {
         response.data?.content?.forEach(item => {
-          set(templateTypes, getScopeBasedTemplateRef(item), parse(item.yaml || '').template.spec.type)
+          const templateData = parse(item.yaml || '').template
+          set(templateTypes, getScopeBasedTemplateRef(item), templateData.spec.type)
+          if (templateData.type === Category.STAGE) {
+            const a = getScopeBasedTemplateRef(item)
```
Comment (yc147) [inline src/modules/70-pipeline/utils/templateUtils.ts:154]: Change this

Comment (PratyushGar) [reply src/modules/70-pipeline/utils/templateUtils.ts]: Addressed.

File: src/modules/70-pipeline/components/PipelineStudio/PipelineTemplateBuilder/TemplatePipelineCanvas/__tests__/TemplatePipelineCanvas.test.tsx (hunk 1 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineTemplateBuilder/TemplatePipelineCanvas/__tests__/TemplatePipelineCanvas.test.tsx:118]: Will add in subsequent PR along with the DeployServiceSpecifications change.

**Review by yc147 — APPROVED** [review]

**Pull request merged**